src/sections/account/api-token-section/ApiTokenSection.tsx

```diff
@@ -2,22 +2,80 @@ import { Trans, useTranslation } from 'react-i18next'
 import { Button } from '@iqss/dataverse-design-system'
 import accountStyles from '../Account.module.scss'
 import styles from './ApiTokenSection.module.scss'
+import { useEffect, useState } from 'react'
+import { Alert } from '@iqss/dataverse-design-system'
+import { TokenInfo } from '@/users/domain/models/TokenInfo'
+import { ApiTokenInfoRepository } from '@/users/domain/repositories/ApiTokenInfoRepository'
+import ApiTokenSectionSkeleton from './ApiTokenSectionSkeleton'
+import { useGetApiToken } from './useGetCurrentApiToken'
+import { useRecreateApiToken } from './useRecreateApiToken'
+import { useRevokeApiToken } from './useRevokeApiToken'
+interface ApiTokenSectionProps {
+  repository: ApiTokenInfoRepository
+}
 
-export const ApiTokenSection = () => {
+export const ApiTokenSection = ({ repository }: ApiTokenSectionProps) => {
   const { t } = useTranslation('account', { keyPrefix: 'apiToken' })
+  const [currentApiTokenInfo, setCurrentApiTokenInfo] = useState<TokenInfo>()
+
+  const { error, apiTokenInfo, isLoading } = useGetApiToken(repository)
+
+  const getError =
+    error !== 'There was an error when reading the resource. Reason was: [404] Token not found.'
```

Comment:
Hi German, thanks for reviewing! 

I put this to distinguish between other errors and not-created token error. If there are other errors, then an error alert would come out `<Alert variant="danger" dismissible={false}>
        {getError || recreatingError || revokingError}
      </Alert>.` Otherwise, the page would say something like "you should create an api token first". 'There was an error when reading the resource. Reason was: [404] Token not found.' this message from API just means we should create an api token, so I filter it out. I'm not sure if we need it.

Reply:
I think you could handle that logic in the same `useGetApiToken` hook where you set the error?